### PR TITLE
[Core][Geometry] Adding functions for normal and unit normal

### PR DESCRIPTION
**Pull request by tteschemacher**
Hi all,

I would like to complete the normal and the unit normal functions in the geometry base class to make them accessable via the integration point index. It is a bit more efficient than calling it with the coordinates, because then the shape functions would not have to be computed each time.

### CODE REVIEW

File: kratos/geometries/geometry.h [hunk 1]
```diff
@@ -832,12 +832,72 @@ class Geometry : public PointerVector<TPointType>
...
+        const SizeType local_space_dimension = this->LocalSpaceDimension();
+        const SizeType dimension = this->WorkingSpaceDimension();
+
+        KRATOS_DEBUG_ERROR_IF(dimension == local_space_dimension)
+            << "Remember the normal can be computed just in geometries with a local dimension: "
+            << this->LocalSpaceDimension() << "smaller than the spatial dimension: "
+            << this->WorkingSpaceDimension() << std::endl;
+
+        // We define the normal and tangents
+        array_1d<double, 3> tangent_xi(3, 0.0);
```
Comment (loumalouomega) [inline kratos/geometries/geometry.h]:
```suggestion
        array_1d<double, 3> tangent_xi = ZeroVector(3);
```
This initialization will stop working with AMatrix

File: kratos/geometries/geometry.h [hunk 2]
```diff
@@ -832,12 +832,72 @@ class Geometry : public PointerVector<TPointType>
...
+        array_1d<double, 3> tangent_eta(3, 0.0);
```
Comment (loumalouomega) [inline kratos/geometries/geometry.h]:
```suggestion
        array_1d<double, 3> tangent_eta = ZeroVector(3);
```

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Minor comments (this should be applied to the old code too)

File: kratos/geometries/geometry.h (hunk 2 above)
Comment (loumalouomega) [reply kratos/geometries/geometry.h]: After this I will approve (I assume @KratosMultiphysics/technical-committee  doesn't have any problem with this)

**Comment by RiccardoRossi** [comment]
the two options are needed...

File: kratos/geometries/geometry.h (hunk 2 above)
Comment (tteschemacher) [reply kratos/geometries/geometry.h]: Good to know. I do this wrong many times...
Thanks!

Comment (tteschemacher) [reply kratos/geometries/geometry.h]: I changed it accordingly.

**Review by loumalouomega — APPROVED** [review]
Approving then

File: kratos/geometries/geometry.h [hunk 3]
```diff
@@ -832,12 +832,72 @@ class Geometry : public PointerVector<TPointType>
...
 
+    /**
+     * @brief It returns the vector, which is normal to its corresponding
+     *        geometry in the given integration point for the default
+     *        integration method.
+     * @param IntegrationPointIndex index in internal integration point list
+     * @return The normal in the given integration point
+     */
+    virtual array_1d<double, 3> Normal(
+        IndexType IntegrationPointIndex) const
```
Comment (RiccardoRossi) [inline kratos/geometries/geometry.h:843]: i would avoid this method and only use the following with a defaukt value

Comment (tteschemacher) [reply kratos/geometries/geometry.h:843]: The default is a member variable, thus I cannot put it as default.

**Pull request merged**